source/containers.tex

```diff
@@ -21425,3 +21490,631 @@
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{Extents} is a specialization of \tcode{extents}.
+
+\pnum
+Let \tcode{index_type} be \tcode{typename Extents::index_type}.
+
+%FIXME: What is $k$ in this itemdescr? Don't we want \tcode{k} and not $k$?
+\pnum
+Let $\lambda{}_k$ denote the following value:
+\begin{itemize}
+\item
+\tcode{\exposid{de-ice}($s_k$) + 1}
+if $S_k$ models \tcode{\libconcept{convertible_to}<index_type>}; otherwise
+\item
+\tcode{get<1>($s_k$)}
+if $S_k$ models \tcode{\exposconcept{index-pair-like}<index_type>}; otherwise
+\item
+\tcode{\exposid{de-ice}($s_k$.offset)} \tcode{+}
+\tcode{\exposid{de-ice}($s_k$.extent)}
+if $S_k$ is a specialization of \tcode{strided_slice}; otherwise
+\item
+%FIXME: Note that the paper uses \tcode{k} here.
+\tcode{src.extent(k)}.
+\end{itemize}
+
+\pnum
+\expects
+$\lambda{}_k$ is representable as a value of type \tcode{index_type}.
+
+\pnum
+\returns
+\tcode{Extents::\exposid{index-cast}($\lambda{}_k$)}.
+\end{itemdescr}
+
+\indexlibraryglobal{\exposid{src-indices}}%
+\begin{itemdecl}
+template<class IndexType, size_t N, class... SliceSpecifiers>
+  constexpr array<IndexType, sizeof...(SliceSpecifiers)>
+    @\exposid{src-indices}@(const array<IndexType, N>& indices, SliceSpecifiers... slices);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{IndexType} is a signed or unsigned integer type.
+
+\pnum
+\returns
+An \tcode{array<IndexType, sizeof...(SliceSpecifiers)> src_idx} such that
+for each $k$ in the range \range{0}{sizeof...(SliceSpecifiers)},
+\tcode{src_idx[$k$]} equals
+
+\begin{itemize}
+\item
+%FIXME: We're redefining what $k$ is here.
+%FIXME: Also, calling first_ on each $k$ below gives us a set of indeces - how do we assign a set to the src_idx[$k$] above?
+\tcode{\exposid{first_}<IndexType, $k$>(slices...)} for each $k$
+where \tcode{\placeholder{map-rank}[$k$]} equals
+\tcode{dynamic_extent},
+
+\item
+%FIXME: How do we fail the previous \item to get here? There's no "if" test to fail.
+otherwise,
+\tcode{\exposid{first_}<IndexType, $k$>(slices...)} \tcode{+}
+\tcode{indices[\placeholder{map-rank}[$k$]]}.
+\end{itemize}
+\end{itemdescr}
+
+\rSec4[mdspan.submdspan.extents]{\tcode{submdspan_extents} function}
+
+\indexlibraryglobal{submdspan_extents}%
+\begin{itemdecl}
+template<class IndexType, class... Extents, class... SliceSpecifiers>
+  constexpr auto submdspan_extents(const extents<IndexType, Extents...>& src,
+                                   SliceSpecifiers... slices);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{sizeof...(slices)} equals \tcode{Extents::rank()}.
+
+\pnum
+\mandates
+For each rank index $k$ of \tcode{src.extents()},
+exactly one of the following is true:
+\begin{itemize}
+\item $S_k$ models \tcode{\libconcept{convertible_to}<IndexType>},
+\item $S_k$ models \tcode{\exposconcept{index-pair-like}<IndexType>},
+\item \tcode{is_convertible_v<$S_k$, full_extent_t>} is \tcode{true}, or
```

Comment:
Where is `full_extent_t` defined?

Reply:
I opened #6353 for this.
P0009 (R18) mentions in "1.8 P0009r11: 2021-05 Mailing"
> Renamed all_type to full_extent_t and all to full_extent

My guess is that `submdspan` wasn't up to date.

Reply:
@crtrott ^^^ ?

Reply:
OK, thanks for the issue! We can defer that if there's no immediate solution at hand.